src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -1162,8 +1162,16 @@ private async Task HandleBreakModeEvent(ResultEventArgs results, BreakRequest br
                 string bkptno = results.Results.FindString("bkptno");
                 ulong addr = cxt.pc ?? 0;
 
+                EXCEPTION_INFO exceptionInfo;
                 bool fContinue;
                 TupleValue frame = results.Results.TryFind<TupleValue>("frame");
+
+                // test -- need to delete
+                if (ExceptionManager.TryGetExceptionBreakpoint(bkptno, out exceptionInfo) == Constants.S_OK)
```

Comment:
You can move this block to line 1200.

The flow would be that if we fail to find it as a regular breakpoint, it would check to see if it is an exception breakpoint (catchpoint).

As of now you are going to fire two events.

You should also be able to declare the type within the out parameter.

```suggestion
                if (ExceptionManager.TryGetExceptionBreakpoint(bkptno, out EXCEPTION_INFO exceptionInfo) == Constants.S_OK)
```